tests/end2end/playwright/dnd-form.spec.js

```diff
@@ -1,89 +1,143 @@
-        await expect(
-            page.locator(
-                '#attribute-table-panel-dnd_form_geom > div.lizmapPopupSingleFeature > div > table > tbody > tr:nth-child(3) > td'
-            )
-        ).not.toBeEmpty();
-
-        // Assert data has not been removed after form submission w/o modification
-        await page
-            .locator('#attribute-layer-table-dnd_form_geom .feature-edit')
-            .click({ force: true });
-        await page.locator('#jforms_view_edition__submit_submit').click();
-        await page.waitForTimeout(300);
-        await expect(
-            page.locator(
-                '#attribute-table-panel-dnd_form_geom > div.lizmapPopupSingleFeature > div > table > tbody > tr:nth-child(2) > td'
-            )
-        ).not.toBeEmpty();
-        await expect(
-            page.locator(
-                '#attribute-table-panel-dnd_form_geom > div.lizmapPopupSingleFeature > div > table > tbody > tr:nth-child(3) > td'
-            )
-        ).not.toBeEmpty();
-
-        // Assert data has changed after form submission w modification
-        await page
-            .locator('#attribute-layer-table-dnd_form_geom .feature-edit')
-            .click({ force: true });
-        await page.locator('#jforms_view_edition-tabs > li:nth-child(2)').click();
-        await page.locator('#jforms_view_edition_field_in_dnd_form').clear();
-        await page
-            .locator('#jforms_view_edition_field_in_dnd_form')
-            .fill('modified');
-        await page.locator('#jforms_view_edition__submit_submit').click();
-
-        // Click on line to refresh popup info
-        await page
-            .locator('#attribute-layer-table-dnd_form_geom tbody tr')
-            .first()
-            .click({ force: true });
-        await expect(
-            page.locator(
-                '#attribute-table-panel-dnd_form_geom > div.lizmapPopupSingleFeature > div > table > tbody > tr:nth-child(2) > td'
-            )
-        ).toHaveText('modified');
-        await expect(
-            page.locator(
-                '#attribute-table-panel-dnd_form_geom > div.lizmapPopupSingleFeature > div > table > tbody > tr:nth-child(3) > td'
-            )
-        ).toHaveText('test_geom');
-
-        // Write back original data (TODO: refresh database data?)
+    test('With non spatial data creation, not remove data', async function ({ page }) {
+        const project = new ProjectPage(page, 'dnd_form');
+        await project.open();
+        await project.openAttributeTable('dnd_form');
+
+        await project.bottomDock.locator(".btn-createFeature-attributeTable").click();
+        await expect(project.dock).toBeVisible();
+
+        await project.dock.getByText('tab2').click();
+        await project.editingField('field_in_dnd_form').fill('value in DND form');
+        await project.editingSubmitForm('close');
+        const ids = await editedFeatureIds(page);
+
+        const table = project.attributeTableHtml('dnd_form');
```

Comment:
```suggestion
        const table = await project.attributeTableHtml('dnd_form');
```
is `attributeTableHtml` an `async` method ?